src/calculate_proof.rs

```diff
@@ -1,20 +1,23 @@
 use crypto_hash::{Algorithm, digest};
 use std::ops::Range;
+use std::str;
 use rayon::prelude::*;
 
 use block::Block;
 use to_json::ToJSON;
 
 fn prove_range(block: &Block, number_of_zeroes: usize, range: Range<u64>) -> Option<Block> {
+    const TEMPLATE_STR: &'static str = "\"proof\":123456789";
+    let template_json = block.with_proof(123456789).to_json();
     for proof in range {
-        let new_block = block.with_proof(proof);
-        let block_as_json = new_block.to_json();
-        let block_sha256 = digest(Algorithm::SHA256, block_as_json.as_bytes());
+        let substr = format!("\"proof\":{}", proof);
```

Comment:
Why is `format` a macro and not a function? Did you write that `format` macro? Is it faster to use a macro?

The rest looks fine. 😀